src/data/AudioProcessor.py

```diff
@@ -0,0 +1,134 @@
+        spectrogram_db = librosa.power_to_db(spectrogram, ref=np.max)
+        return spectrogram_db
+
+    def create_mel_spectrogram(self, file_path, n_mels=128, hop_length=512):
+        y, sr = librosa.load(file_path, sr=self.sample_rate)
+        stft = librosa.stft(y)
+        mel_spectrogram = librosa.feature.melspectrogram(
+            S=np.abs(stft), sr=sr, n_mels=n_mels, hop_length=hop_length
+        )
+        db_mel_spectrogram = librosa.amplitude_to_db(mel_spectrogram, ref=np.max)
+        return db_mel_spectrogram
+
+    def save_mel_spectrogram(self, mel_spectrogram_db, target_path, sr, hop_length):
+        plt.figure(figsize=(10, 5))
+        librosa.display.specshow(
+            mel_spectrogram_db,
+            sr=sr,
+            x_axis="time",
+            y_axis="mel",
+            hop_length=hop_length,
+        )
+        plt.colorbar()
+        plt.title(target_path.split("/")[-1])
+        plt.tight_layout()
+
+        plot_path = os.path.join(
+            self.spectrogram_directory,
+            target_path.split("/")[-2],
+            target_path.split("/")[-1],
+        )
+        plot_path = plot_path.replace(".mp3", ".png")
+        plt.savefig(plot_path)
+        plt.close()
+
+    def plot_mel_spectrogram(self, mel_spectrogram_db, target_path, sr, hop_length):
+        plt.figure(figsize=(10, 5))
+        librosa.display.specshow(
+            mel_spectrogram_db,
+            sr=sr,
+            x_axis="time",
+            y_axis="mel",
+            hop_length=hop_length,
+        )
+        plt.colorbar()
+        plt.title(target_path.split("/")[-1])
+        plt.tight_layout()
+        plt.show()
+        plt.close()
+
+    def process_directory(self):
+        for dirpath, directories, _ in os.walk(self.source_directory):
+            for directory in directories:
+                dirpath = os.path.join(dirpath, directory)
+                # check if required directories exist, if not then create them
+                os.makedirs(
+                    os.path.join(self.spectrogram_directory, directory), exist_ok=True
+                )
+                os.makedirs(
+                    os.path.join(self.target_directory, directory), exist_ok=True
+                )
+                for file in os.listdir(dirpath):
+                    try:
+                        file_path = os.path.join(dirpath, file)
+                        target_file_path = self.preprocess_audio(file_path)
+                        if target_file_path is not None:
+                            ms = self.create_mel_spectrogram(target_file_path)
+                            self.save_mel_spectrogram(
+                                ms, target_file_path, self.sample_rate, hop_length=512
+                            )
+                    except Exception as e:
+                        print(f"Error processing file {file_path}: {e}")
+                    # lets test only for 1 file
```

Comment:
What does it mean? Does this comment suggests that the code actually tests just one file?